src/components/dialog/dialog.pw.tsx

```diff
@@ -0,0 +1,596 @@
+        mount,
+        page,
+      }) => {
+        await mount(<DialogWithToast />);
+        const dialog = page.getByRole("dialog");
+        const openToastButton = page
+          .getByRole("button")
+          .filter({ hasText: "Open Toast" });
+        await openToastButton.click();
+
+        await page.mouse.click(0, 0); // click outside Toast and Dialog
+        await dialog.press("Tab");
+
+        await expect(openToastButton).toBeFocused();
+      });
+
+      test("when tabbing through Dialog content and two opened Toasts, the background scroll should not scroll to the bottom of the page", async ({
+        mount,
+        page,
+      }) => {
+        await mount(<DialogWithOpenToastsBackgroundScrollTest />);
+        const dialog = page.getByRole("dialog");
+        const dialogCloseButton = dialog.getByLabel("Close");
+        const textbox = page.getByLabel("Textbox");
+        const toast1CloseButton = toastComponent(page)
+          .filter({ hasText: "Toast message 1" })
+          .getByLabel("Close");
+        const toast2CloseButton = toastComponent(page)
+          .filter({ hasText: "Toast message 2" })
+          .getByLabel("Close");
+
+        await dialog.press("Tab");
+        await dialogCloseButton.press("Tab");
+        await textbox.press("Tab");
+        await toast1CloseButton.press("Tab");
+        await toast2CloseButton.press("Tab");
+
+        await expect(dialogCloseButton).toBeFocused();
+        await expect(
+          page.getByText("I should not be scrolled into view")
+        ).not.toBeInViewport();
+      });
+
+      test("when shift tabbing through Dialog content and two opened Toasts, the background scroll should not scroll to the bottom of the page", async ({
+        mount,
+        page,
+      }) => {
+        await mount(<DialogWithOpenToastsBackgroundScrollTest />);
+        const dialog = page.getByRole("dialog");
+        const dialogCloseButton = dialog.getByLabel("Close");
+        const textbox = page.getByLabel("Textbox");
+        const toast1CloseButton = page
+          .locator('[data-component="toast"]')
+          .filter({ hasText: "Toast message 1" })
+          .getByLabel("Close");
+        const toast2CloseButton = page
+          .locator('[data-component="toast"]')
+          .filter({ hasText: "Toast message 2" })
+          .getByLabel("Close");
+
+        await dialog.press("Shift+Tab");
+        await toast2CloseButton.press("Shift+Tab");
+        await toast1CloseButton.press("Shift+Tab");
+        await textbox.press("Shift+Tab");
+
+        await expect(dialogCloseButton).toBeFocused();
+        await expect(
+          page.getByText("I should not be scrolled into view")
+        ).not.toBeInViewport();
+      });
+    }
+  );
+
+  test("Dialog should have rounded corners", async ({ mount, page }) => {
+    await mount(<DialogComponent />);
```

Comment:
suggestion: (non-blocking) it would be better to add a space to separate the mount from the assertion to follow the same pattern as with the other test cases.